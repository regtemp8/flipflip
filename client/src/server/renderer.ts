import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
