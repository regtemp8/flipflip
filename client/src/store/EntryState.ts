export interface Identifiable {
  id: number
}

export interface EntryState<T extends Identifiable> {
  name: string
  nextID: number
  entries: Record<number, T>
}

export interface EntryUpdate<T> {
  id: number
  value: T
}

export function isEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  return state.entries[id] != null
}

export function setNextID<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (state.nextID <= entry.id) {
    state.nextID = entry.id + 1
  }
}

export function getEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
): T {
  return state.entries[id]
}

export function setEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  state.entries[entry.id] = entry
  setNextID(state, entry)
}

export function createEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (!isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
    setNextID(state, entry)
  } else {
    throw new Error(
      `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
    )
  }
}

export function createEntries<T extends Identifiable>(
  state: EntryState<T>,
  entries: T[]
) {
  for (const entry of entries) {
    if (isEntry(state, entry.id)) {
      throw new Error(
        `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
      )
    }
  }

  entries.forEach((s) => (state.entries[s.id] = s))
  const maxID = Math.max(...entries.map((s) => s.id))
  setNextID(state, { id: maxID })
}

export function updateEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
  } else {
    throw new Error(
      `${state.name}: Update failed. Entry with ID ${entry.id} doesn't exist.`
    )
  }
}

export function deleteEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  if (isEntry(state, id)) {
    delete state.entries[id]
  } else {
    throw new Error(
      `${state.name}: Delete failed. Entry with ID ${id} doesn't exist.`
    )
  }
}
