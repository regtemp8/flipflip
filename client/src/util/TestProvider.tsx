import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
