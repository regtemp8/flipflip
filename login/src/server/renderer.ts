import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
