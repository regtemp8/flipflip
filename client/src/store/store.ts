/// <reference path="../global.d.ts"/>
import { createReduxStore } from '../server/renderer'

const store = createReduxStore(
  window.flipflipAppStorage,
  window.flipflipConstants
)

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
