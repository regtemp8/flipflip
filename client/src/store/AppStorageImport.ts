import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { EntryState } from "./EntryState";
import type Tag from "./tag/Tag";
import type SceneGroup from "./sceneGroup/SceneGroup";
import type SceneGrid from "./sceneGrid/SceneGrid";
import type Scene from "./scene/Scene";
import type Playlist from "./playlist/Playlist";
import type Overlay from "./overlay/Overlay";
import type LibrarySource from "./librarySource/LibrarySource";
import type Clip from "./clip/Clip";
import type CaptionScript from "./captionScript/CaptionScript";
import type Audio from "./audio/Audio";
import type App from "./app/data/App";

export interface AppStorageImport {
  app: App;
  audio: EntryState<Audio>;
  captionScript: EntryState<CaptionScript>;
  clip: EntryState<Clip>;
  librarySource: EntryState<LibrarySource>;
  overlay: EntryState<Overlay>;
  playlist: EntryState<Playlist>;
  scene: EntryState<Scene>;
  sceneGrid: EntryState<SceneGrid>;
  sceneGroup: EntryState<SceneGroup>;
  tag: EntryState<Tag>;
}

export const initialAppStorageImport: AppStorageImport = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
}