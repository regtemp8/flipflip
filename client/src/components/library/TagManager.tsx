import React, { ChangeEvent, MouseEvent, useState } from 'react'
import { cx } from '@emotion/css'
import Sortable from 'react-sortablejs'

import {
  AppBar,
  Button,
  Card,
  CardActionArea,
  CardContent,
  Container,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Fab,
  IconButton,
  ListItemSecondaryAction,
  ListItemText,
  Menu,
  MenuItem,
  TextField,
  type Theme,
  Toolbar,
  Tooltip,
  Typography
} from '@mui/material'

import { makeStyles } from 'tss-react/mui'

import AddIcon from '@mui/icons-material/Add'
import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward'
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward'
import DeleteIcon from '@mui/icons-material/Delete'
import DeleteSweepIcon from '@mui/icons-material/DeleteSweep'
import SortIcon from '@mui/icons-material/Sort'

import { en, MO, SF } from 'flipflip-common'
import { arrayMove } from '../../data/utils'
import Jiggle from '../../animations/Jiggle'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import {
  removeAllTags,
  setRouteGoBack,
  sortTags,
  saveTag
} from '../../store/app/thunks'
import { newTag } from '../../store/tag/Tag'
import { selectAppTags } from '../../store/app/selectors'
import { selectTagName, selectTagNextID } from '../../store/tag/selectors'

const useStyles = makeStyles()((theme: Theme) => ({
  root: {
    display: 'flex'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  appBarSpacer: {
    backgroundColor: theme.palette.primary.main,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    minHeight: 64
  },
  backButton: {
    float: 'left'
  },
  title: {
    textAlign: 'center',
    flexGrow: 1
  },
  headerBar: {
    display: 'flex',
    alignItems: 'center',
    whiteSpace: 'nowrap',
    flexWrap: 'nowrap'
  },
  headerLeft: {
    flexBasis: '3%'
  },
  content: {
    display: 'flex',
    flexGrow: 1,
    flexDirection: 'column',
    height: '100vh',
    backgroundColor: theme.palette.background.default
  },
  container: {
    padding: theme.spacing(0),
    overflowY: 'auto'
  },
  tagList: {
    padding: theme.spacing(1),
    display: 'flex',
    flexWrap: 'wrap'
  },
  tag: {
    marginRight: theme.spacing(1),
    marginBottom: theme.spacing(1)
  },
  addMenuButton: {
    backgroundColor: theme.palette.primary.dark,
    margin: 0,
    top: 'auto',
    right: 20,
    bottom: 20,
    left: 'auto',
    position: 'fixed'
  },
  sortMenuButton: {
    backgroundColor: theme.palette.secondary.dark,
    margin: 0,
    top: 'auto',
    right: 80,
    bottom: 20,
    left: 'auto',
    position: 'fixed'
  },
  removeAllButton: {
    backgroundColor: theme.palette.error.main,
    margin: 0,
    top: 'auto',
    right: 130,
    bottom: 20,
    left: 'auto',
    position: 'fixed'
  },
  icon: {
    color: theme.palette.primary.contrastText
  },
  sortMenu: {
    width: 200
  },
  fill: {
    flexGrow: 1
  },
  phraseInput: {
    minWidth: 200,
    minHeight: 100
  }
}))

interface TagEditDialogProps {
  tagID: number
  onClose: () => void
}

function TagEditDialog(props: TagEditDialogProps) {
  const dispatch = useAppDispatch()
  const tagNextID = useAppSelector(selectTagNextID())

  const [tagName, setTagName] = useState('')
  const [tagPhrase, setTagPhrase] = useState('')

  const onChangeTitle = (e: ChangeEvent<HTMLInputElement>) => {
    setTagName(e.currentTarget.value)
  }

  const onChangePhrase = (e: ChangeEvent<HTMLInputElement>) => {
    setTagPhrase(e.currentTarget.value)
  }

  const onRemoveTag = () => {
    setTagName('')
    setTagPhrase('')
    props.onClose()
  }

  const onFinishEdit = () => {
    dispatch(
      saveTag(
        newTag({ id: props.tagID, name: tagName, phraseString: tagPhrase })
      )
    )
    setTagName('')
    setTagPhrase('')
    props.onClose()
  }

  const onCloseEditDialog = () => {
    if (props.tagID === tagNextID && tagName === '') {
      onRemoveTag()
    } else {
      props.onClose()
    }
  }

  return (
    <Dialog
      open={props.tagID !== -1}
      onClose={onCloseEditDialog}
      aria-labelledby="edit-title"
    >
      <DialogTitle id="edit-title">Edit Tag</DialogTitle>
      <DialogContent>
        <TextField
          variant="standard"
          autoFocus
          fullWidth
          required
          label="Name"
          value={tagName}
          margin="dense"
          onChange={onChangeTitle}
        />
        <TextField
          variant="standard"
          fullWidth
          multiline
          label="Tag Phrases"
          helperText="These are used in place of $TAG_PHRASE for Caption scripts. One per line."
          id="phrase"
          value={tagPhrase}
          margin="dense"
          inputProps={{ className: props.classes.phraseInput }}
          onChange={onChangePhrase}
        />
      </DialogContent>
      <DialogActions>
        <IconButton
          onClick={onRemoveTag}
          style={{ marginRight: 'auto' }}
          size="large"
        >
          <DeleteIcon color="error" />
        </IconButton>
        <Button onClick={onCloseEditDialog} color="secondary">
          Cancel
        </Button>
        <Button disabled={!tagName} onClick={onFinishEdit} color="primary">
          OK
        </Button>
      </DialogActions>
    </Dialog>
  )
}

interface TagCardProps {
  tagID: number
  onEdit: (id: number) => void
}

function TagCard(props: TagCardProps) {
  const name = useAppSelector(selectTagName(props.tagID)) || ''

  return (
    <Jiggle key={props.tagID + name} bounce>
      <Card className={props.classes.tag}>
        <CardActionArea onClick={() => props.onEdit(props.tagID)}>
          <CardContent>
            <Typography component="h2" variant="h6">
              {name}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </Jiggle>
  )
}

function TagManager() {
  const dispatch = useAppDispatch()
  const tagNextID = useAppSelector(selectTagNextID())
  const tags = useAppSelector(selectAppTags())
  const [openMenu, setOpenMenu] = useState<string>()
  const [menuAnchorEl, setMenuAnchorEl] = useState<any>()
  const [isEditing, setIsEditing] = useState(-1)

  const goBack = () => {
    dispatch(setRouteGoBack())
  }

  const onCloseDialog = () => {
    setMenuAnchorEl(null)
    setOpenMenu(undefined)
    setIsEditing(-1)
  }

  const onAddTag = () => {
    setIsEditing(tagNextID)
  }

  const onEditTag = (tagID: number) => {
    setIsEditing(tagID)
  }

  const onRemoveAll = () => {
    setOpenMenu(MO.removeAllAlert)
  }

  const onOpenSortMenu = (e: MouseEvent<HTMLButtonElement>) => {
    setMenuAnchorEl(e.currentTarget)
    setOpenMenu(MO.sort)
  }

  const onFinishRemoveAll = () => {
    dispatch(removeAllTags())
    onCloseDialog()
  }

  const { classes } = useStyles()
  return (
    <div className={classes.root}>
      <AppBar enableColorOnDark position="absolute" className={classes.appBar}>
        <Toolbar className={classes.headerBar}>
          <div className={classes.headerLeft}>
            <Tooltip disableInteractive title="Back" placement="right-end">
              <IconButton
                edge="start"
                color="inherit"
                aria-label="Back"
                className={classes.backButton}
                onClick={goBack}
                size="large"
              >
                <ArrowBackIcon />
              </IconButton>
            </Tooltip>
          </div>

          <Typography
            component="h1"
            variant="h4"
            color="inherit"
            noWrap
            className={classes.title}
          >
            Tag Manager
          </Typography>

          <div className={classes.headerLeft} />
        </Toolbar>
      </AppBar>

      <main className={classes.content}>
        <div className={classes.appBarSpacer} />

        <div className={cx(classes.root, classes.fill)}>
          <Container maxWidth={false} className={classes.container}>
            <Sortable
              className={classes.tagList}
              options={{
                animation: 150,
                easing: 'cubic-bezier(1, 0, 0, 1)'
              }}
              onChange={(order: any, sortable: any, evt: any) => {
                arrayMove(tags, evt.oldIndex, evt.newIndex)
              }}
            >
              {tags.map((tagID) => (
                <TagCard tagID={tagID} onEdit={onEditTag} classes={classes} />
              ))}
            </Sortable>
          </Container>
        </div>
        <TagEditDialog
          tagID={isEditing}
          onClose={onCloseDialog}
          classes={classes}
        />
      </main>

      {tags.length > 0 && (
        <React.Fragment>
          <Tooltip disableInteractive title="Remove All Tags">
            <Fab
              className={classes.removeAllButton}
              onClick={onRemoveAll}
              size="small"
            >
              <DeleteSweepIcon className={classes.icon} />
            </Fab>
          </Tooltip>
          <Dialog
            open={openMenu === MO.removeAllAlert}
            onClose={onCloseDialog}
            aria-labelledby="remove-all-title"
            aria-describedby="remove-all-description"
          >
            <DialogTitle id="remove-all-title">Delete Tags</DialogTitle>
            <DialogContent>
              <DialogContentText id="remove-all-description">
                Are you sure you want to remove all Tags? This will untag all
                sources as well.
              </DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button onClick={onCloseDialog} color="secondary">
                Cancel
              </Button>
              <Button onClick={onFinishRemoveAll} color="primary">
                OK
              </Button>
            </DialogActions>
          </Dialog>
        </React.Fragment>
      )}

      {tags.length >= 2 && (
        <React.Fragment>
          <Fab
            className={classes.sortMenuButton}
            aria-haspopup="true"
            aria-controls="sort-menu"
            aria-label="Sort Tags"
            onClick={onOpenSortMenu}
            size="medium"
          >
            <SortIcon className={classes.icon} />
          </Fab>
          <Menu
            id="sort-menu"
            elevation={1}
            anchorOrigin={{
              vertical: 'top',
              horizontal: 'center'
            }}
            transformOrigin={{
              vertical: 'bottom',
              horizontal: 'right'
            }}
            anchorEl={menuAnchorEl}
            keepMounted
            classes={{ paper: classes.sortMenu }}
            open={openMenu === MO.sort}
            onClose={onCloseDialog}
          >
            {[SF.alpha, SF.date].map((sf) => (
              <MenuItem key={sf}>
                <ListItemText primary={en.get(sf)} />
                <ListItemSecondaryAction>
                  <IconButton
                    edge="end"
                    onClick={() => {
                      dispatch(sortTags(sf, true))
                    }}
                    size="large"
                  >
                    <ArrowUpwardIcon />
                  </IconButton>
                  <IconButton
                    edge="end"
                    onClick={() => {
                      dispatch(sortTags(sf, false))
                    }}
                    size="large"
                  >
                    <ArrowDownwardIcon />
                  </IconButton>
                </ListItemSecondaryAction>
              </MenuItem>
            ))}
          </Menu>
        </React.Fragment>
      )}

      <Fab className={classes.addMenuButton} onClick={onAddTag} size="large">
        <AddIcon className={classes.icon} />
      </Fab>
    </div>
  )
}

;(TagManager as any).displayName = 'TagManager'
export default TagManager
