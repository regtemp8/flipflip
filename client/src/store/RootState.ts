import { RootState } from "./store";
import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialSystemConstants } from "./constants/SystemConstants";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { initialPlayersState } from "./player/slice";
import { initialVideoClipperState } from "./videoClipper/slice";
import { initialScenePickerState } from "./scenePicker/slice";
import { initialCaptionScriptorState } from "./captionScriptor/slice";
import { initialSceneDetailState } from "./sceneDetail/slice";
import { initialAudioLibraryState } from "./audioLibrary/slice";
import { initialSourceScraperState } from "./sourceScraper/slice";

export const initialRootState: RootState = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  constants: initialSystemConstants,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
  players: initialPlayersState,
  videoClipper: initialVideoClipperState,
  scenePicker: initialScenePickerState,
  captionScriptor: initialCaptionScriptorState,
  sceneDetail: initialSceneDetailState,
  audioLibrary: initialAudioLibraryState,
  sourceScraper: initialSourceScraperState
}