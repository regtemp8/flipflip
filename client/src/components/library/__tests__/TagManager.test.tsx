import React from 'react'
import { describe, it, expect } from '@jest/globals'
import renderer from 'react-test-renderer'
import TagManager from '../TagManager'
import TestProvider from '../../../util/TestProvider'
import store from '../../../store/store'

jest.mock('../../../animations/Jiggle', () => 'Jiggle')

// mocking this so that test doesn't throw error
jest.mock('react-sortablejs', () => 'Sortable')

describe('TagManager', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <TagManager />
      </TestProvider>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
