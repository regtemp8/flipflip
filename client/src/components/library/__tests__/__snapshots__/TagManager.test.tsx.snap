// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagManager should match snapshot 1`] = `
<div
  className="TagManager-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionAbsolute TagManager-appBar-2 css-jz7rmq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular TagManager-headerBar-6 css-hyum1k-MuiToolbar-root"
    >
      <div
        className="TagManager-headerLeft-7"
      >
        <button
          aria-label="Back"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge TagManager-backButton-4 css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap TagManager-title-5 css-1r3v6n7-MuiTypography-root"
      >
        Tag Manager
      </h1>
      <div
        className="TagManager-headerLeft-7"
      />
    </div>
  </header>
  <main
    className="TagManager-content-8"
  >
    <div
      className="TagManager-appBarSpacer-3"
    />
    <div
      className="TagManager-root-1 TagManager-fill-17"
    >
      <div
        className="MuiContainer-root TagManager-container-9 css-ur2jdm-MuiContainer-root"
      >
        <Sortable
          className="TagManager-tagList-10"
          onChange={[Function]}
          options={
            Object {
              "animation": 150,
              "easing": "cubic-bezier(1, 0, 0, 1)",
            }
          }
        />
      </div>
    </div>
  </main>
  <button
    className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default TagManager-addMenuButton-12 css-mm42gz-MuiButtonBase-root-MuiFab-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium TagManager-icon-15 css-i4bv87-MuiSvgIcon-root"
      data-testid="AddIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
